utils/utils.py

```diff
@@ -18,14 +18,59 @@
 )
 from utils.visualizer import Visualizer
 
+import re
+
+
+def parse_slurm_nodelist(nodelist):
+    """
+    Parse SLURM_NODELIST env string to get list of nodes.
+    Usage example:
+        parse_slurm_nodelist(os.environ["SLURM_NODELIST"])
+    Input examples:
+        "or-condo-g04"
+        "or-condo-g[05,07-08,13]"
+        "or-condo-g[05,07-08,13],or-condo-h[01,12]"
+    """
+    nlist = list()
+    for block, _ in re.findall(r"([\w-]+(\[[\d\-,]+\])*)", nodelist):
+        m = re.match(r"^(?P<prefix>[\w\-]+)\[(?P<group>.*)\]", block)
+        if m is None:
+            ## single node
+            nlist.append(block)
+        else:
+            ## multiple nodes
+            g = m.groups()
+            prefix = g[0]
+            for sub in g[1].split(","):
+                if "-" in sub:
+                    start, end = re.match(r"(\d+)-(\d+)", sub).groups()
+                    fmt = "%%0%dd" % (len(start))
+                    for i in range(int(start), int(end) + 1):
+                        node = prefix + fmt % i
+                        nlist.append(node)
+                else:
+                    node = prefix + sub
+                    nlist.append(node)
+
+    return nlist
+
 
 def get_comm_size_and_rank():
-    world_size = 1
+    world_size = None
     world_rank = 0
-    try:
-        world_size = os.environ["OMPI_COMM_WORLD_SIZE"]
-        world_rank = os.environ["OMPI_COMM_WORLD_RANK"]
-    except KeyError:
+
+    if os.getenv("OMPI_COMM_WORLD_SIZE") and os.getenv("OMPI_COMM_WORLD_RANK"):
+        ## Summit
+        world_size = int(os.environ["OMPI_COMM_WORLD_SIZE"])
+        world_rank = int(os.environ["OMPI_COMM_WORLD_RANK"])
+    elif os.getenv("SLURM_NPROCS") and os.getenv("SLURM_PROCID"):
```

Comment:
Is there something different about Slurm that means we can't rely on the MPI distribution variables? I assume you had to add this to work on CADES even though they have OpenMPI, but I hoped we would just need a new option for MPICH etc, not different schedulers